src/gazebo_mavlink_interface.cpp

```diff
@@ -271,6 +271,7 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
   if (_sdf->HasElement("mavlink_udp_port")) {
     mavlink_udp_port_ = _sdf->GetElement("mavlink_udp_port")->Get<int>();
   }
+  model_param(world_->GetName(), model_->GetName(), "mavlink_udp_port", mavlink_udp_port_);
```

Comment:
So are we overriding the `model_udp_port_` value here? Why not put this as the `else` clause of https://github.com/px4/sitl_gazebo/blob/master/src/gazebo_mavlink_interface.cpp#L271-L273?

Reply:
```c++
  if (_sdf->HasElement("mavlink_udp_port")) {
    mavlink_udp_port_ = _sdf->GetElement("mavlink_udp_port")->Get<int>();
  }
  else
     model_param(world_->GetName(), model_->GetName(), "mavlink_udp_port", mavlink_udp_port_);
```

Reply:
We can't. If we using options file with own parameters for each model instance (with different names) with one base model sdf file, we must overwrite mavlink_udp_port from sdf even if it exists. If we will not overwrite, all model instances will use same conflicting port.